### PR TITLE
fix: testing instructions dark theme support

### LINKED ISSUE #9399: [a11y] hover/active state on "test mode" instructions
### Describe the bug

"Test mode" instructions are missing hover/active states when hovering/clicking (respectively) on the button to copy the test number.
"Test mode" instructions are also missing dark theme support.

### To Reproduce

1. Enable "test mode"
2. Go to checkout
3. Click on the card payment method

### Actual behavior

Hover/click states on the clickable test number are not present.

### Screenshots

359816988-2d3e32eb-f476-44ef-98d4-16190ac00e3f (1)

### Expected behavior

359859743-0f4d45c4-2ac4-45b6-826f-313a54caa554

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/9399

#### Changes proposed in this Pull Request

Adding dark theme support for the testing instructions.
Otherwise the "copy" icon look very legible with a dark theme.

Before:
Screenshot 2024-08-21 at 10 30 19 AM

After:
Screenshot 2024-08-21 at 9 58 21 AM

I also tested it with a few variations, on both shortcode and blocks checkout:

Screenshot 2024-08-21 at 10 26 45 AM
Screenshot 2024-08-21 at 10 00 31 AM
Screenshot 2024-08-21 at 10 00 16 AM
Screenshot 2024-08-21 at 9 59 10 AM
Screenshot 2024-08-21 at 9 58 48 AM

#### Testing instructions

I used the Twenty-Twenty-Four theme, which allows to switch between different color palettes.
- Use the TT4 theme
- Comment out these lines, so that the transients are constantly refreshed as you switch color pallettes: https://github.com/Automattic/woocommerce-payments/blob/cd0e4b3c030781fbfd51f5848e5007065349a645/includes/class-wc-payments-checkout.php#L227-L237
- Go to the theme's customizer, to change color palettes: https://wcpay.test/wp-admin/site-editor.php?path=%2Fwp_global_styles
- Go to checkout (both shortcode and blocks)
- The copy icon should be slightly more legible when changing theme colors

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9295` or branch name `fix/testing-instructions-dark-theme-support` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/testing-instructions-dark-theme-support) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6ca3414385ea9c6acba04f889f6e009404bb9393
- Build time: 2024-08-26 08:06:39 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/checkout/classic/payment-processing.js [hunk 1]
```diff
@@ -214,13 +214,20 @@ async function createStripePaymentElement(
 ) {
 	const amount = Number( getUPEConfig( 'cartTotal' ) );
 	const paymentMethodTypes = getPaymentMethodTypes( paymentMethodType );
+	const appearance = await initializeAppearance( api, elementsLocation );
+	document
+		.querySelector(
+			`.wcpay-upe-element[data-payment-method-type="${ paymentMethodType }"]`
+		)
+		?.closest( '.wc_payment_method' )
+		?.classList.add( `theme--${ appearance.theme || 'stripe' }` );
```
Comment (frosso) [inline client/checkout/classic/payment-processing.js:223]: This was a bit unfortunate - having to navigate the DOM. But I guess that's the shortcode checkout way of things.

It's worth mentioning that the `.theme--{themeName}` class name isn't applied to the same element on blocks and shortcode. But it doesn't really matter - it just needs to be applied somewhere in the hierarchy.
The structure of blocks and shortcode checkout is different, anyways - so it's be hard to be consistent.

File: client/checkout/style.scss [hunk 2]
```diff
@@ -26,4 +26,10 @@
 	&:active i {
 		transform: scale( 0.9 );
 	}
+
+	.theme--night & {
+		i {
+			filter: invert( 100% ) hue-rotate( 180deg );
```
Comment (frosso) [inline client/checkout/style.scss:33]: I applied this approach: PCYsg-KnD-p2#dark-mode-via-invert

**Review by FangedParakeet — COMMENTED** [review]
This looks fine to me. Unfortunately, I did notice a few oddities that are mostly unrelated to the changes, but related enough to dark theme support that I thought I'd mention them.  The icon styling appears to be working and appearing fine for me. However, the _button_ styling was doing some strange things on my site.

First of all, I found that the button inherited a white background, until I hovered over it. 

White background on button
_That blinding white!_

This was indeed inherited from my theme.

Theme styles
_The culprit_

This is less important and possibly less offending without the background, but I also noticed a dashed outline on the button on focus.

Dashed outline
_In focus_

This, too, was inherited from the theme. 

Focus styles
_J'accuse_

The outline is less irritating, because it does somewhat surround and cling to the icon, so perhaps not such an eyesore. Nonetheless, while we're here and since I do feel that buttons inheriting hideous styles is a commonplace enough concern, do you think you'd mind adding a few cheeky styling overrides for our humble button here to help correct this? 

(My current theme is Twenty Twenty-One that has been customised intentionally calamitously, FWIW.)

Let me know if you have any thoughts or feel this could be handled in some other way. Cheers!

**Comment by frosso** [comment]
@FangedParakeet thank you for the assessment! I hadn't gone as far back as TT1 during my testing 😅 

> The outline is less irritating, because it does somewhat surround and cling to the icon, so perhaps not such an eyesore. Nonetheless, while we're here and since I do feel that buttons inheriting hideous styles is a commonplace enough concern, do you think you'd mind adding a few cheeky styling overrides for our humble button here to help correct this?

I agree with your analysis - I did a background color override in c19f718e6 , please let me know if that looks a bit better!

**Review by FangedParakeet — APPROVED** [review]
Thanks for the teeny touch! LGTM and works (almost) as expected. 🙌 

I _did_ notice one tinier issue while testing, but it's not technically related to this issue, so I think this one can still be merged happily. Have created #9330 which will explain what I discovered and, if you don't mind us ping-ponging reviews back-and-forth, I have requested your review again on the other PR. 🙏

**Pull request merged**